core/vnl/vnl_matrix.hxx

```diff
@@ -265,10 +265,8 @@ vnl_matrix<T>& vnl_matrix<T>::operator=(vnl_matrix<T>&& rhs)
     else
     {
       // Release any resource we're previously holding
-      if(this->data)
-      {
-        vnl_c_vector<T>::deallocate(this->data[0], this->num_rows*this->num_cols);
-      }
+      vnl_matrix_free_blah;
```

Comment:
@blowekamp, @dzenanz  This is the key change that fixes the bug reported in https://github.com/InsightSoftwareConsortium/ITK/issues/1438

The other changes were necessary to enable tracking this down because the builds were not refecting this code change due to lack of rebuild when this file was changed.